### PR TITLE
add `--parallel` option to `make_docs.py`

### PR DESCRIPTION
**Pull request by yxlao**
This PR:
- Adds `--parallel` option to `make_docs.py`
- Standardizes empty lines in docs:`\n\n\n` -> `\n\n`

Example usage:

```bash
python make_docs.py --parallel --clean_notebooks --execute_notebooks=never --sphinx
```

Merge this PR before merging https://github.com/isl-org/Open3D/pull/4898. 

---
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d/5029)

### CODE REVIEW

**Comment by reyanshsolis** [comment]
LGTM. Testing locally.

**Comment by reyanshsolis** [comment]
I can verify that it is using all cores when using `--parallel`.

With `--execute_notebooks=never`, with `--parallel` it takes about `25sec.`, and without `--parallel` it takes about `36sec.`, on an 8 core / 16 thread system. 

I have added the `--parallel` flag in the `Building the documentation` doc.

File: docs/make_docs.py [hunk 1]
```diff
@@ -631,6 +639,13 @@ def run(self):
         default=False,
         help="Show Open3D version number rather than git hash.",
     )
+    parser.add_argument(
+        "--parallel",
+        action="store_true",
+        default=False,
```
Comment (reyanshsolis) [inline docs/make_docs.py:645]: Do we want to make the default `True`?

**Review by reyanshsolis — APPROVED** [review]

File: docs/make_docs.py (hunk 1 above)
Comment (yxlao) [reply docs/make_docs.py:645]: By default, we may not need the `--parallel` flag, as the error message will be hard to locate. The `--parallel` flag is only used when you know that the compilation will not have errors and you'll recompile the docs again and again.

Comment (yxlao) [reply docs/make_docs.py:645]: I added several examples instead

```bash
    # Run `python make_docs.py --help` to usage of the flags.
    python make_docs.py --help

    # Example: build .rst and C++ docs only, skip notebooks.
    python make_docs.py --execute_notebooks=never --sphinx --doxygen

    # Example: build .rst and C++ docs only, skip notebooks, with parallel build.
    python make_docs.py --execute_notebooks=never --sphinx --doxygen --parallel

    # Example: build .rst and c++ docs, execute notebooks when it has not been executed.
    python make_docs.py --execute_notebooks=auto --sphinx --doxygen
```

**Pull request merged**